cassiopeia/solver/CCPhyloSolver.py

```diff
@@ -0,0 +1,223 @@
+"""
+This file stores a subclass of CassiopeiaSolver, the CCPhyloSolver. This is 
+a wrapper around optimized agglomerative algorithms implemented by CCPhylo
+(https://bitbucket.org/genomicepidemiology/ccphylo/src/master/). Methods
+that will inherit from this class by default are DynamicNeighborJoiningSolver,
+HeuristicNeighborJoiningSolver, NeighborJoiningSolver and UPGMASolver.
+"""
+
+import abc
+from typing import Callable, Dict, List, Optional, Tuple
+
+import networkx as nx
+import numpy as np
+import os
+import tempfile
+import ete3
+import subprocess
+import pandas as pd
+import configparser
+
+from cassiopeia.data import CassiopeiaTree, utilities
+from cassiopeia.mixins import DistanceSolverError
+from cassiopeia.solver import (
+    DistanceSolver,
+    dissimilarity_functions,
+    solver_utilities,
+)
+
+
+class CCPhyloSolver(DistanceSolver.DistanceSolver):
+    """
+    Distance based solver class. This solver serves as a wrapper around 
+    CCPhylo algorithms.
+
+    Args:
```

Comment:
We'll have to add descriptions for the two new parameters here - `fast` and `method`.  We should also be clear about which methods are accepted here, I think we can use the `Literal` type here. (https://docs.python.org/3/library/typing.html#typing.Literal)

Reply:
I'm going to stick with str for this with subsequent value checking so I can raise a more informative error.